### PR TITLE
Remove consecutive convex hull vertices that are very close to each other

### PR DESCRIPTION
**Pull request by mcara**
This PR is very similar to #146 but implements removal of consecutive (adjacent) vertices in the computed convex hull that are "almost" indistinguishable from each other.

This may be needed when accuracy is lost during WCS transformations from image to world coordinates. In those cases, even though the points in the image catalog are different, their world coordinates may be the same.

This PR is different from #146 in the following:
1. filtering is performed on vertices in image coordinates before they are converted to world coordinates. This will provide consistent results regardless of what coordinates are used.
2. points are dropped only when they are close **only to the previous vertex**;
3. test is performed using a tolerance (`min_separation` parameter) and not using exact equality of floating point numbers.

Point number 1 above is important, e.g., for the `RefImage` class which uses custom projection to tangent plane computations and may result in similar issues (loss of accuracy) as regular (i.e., WCSLIB) WCS  transformations and therefore raise exceptions when constructing `SphericalPolygon` from these vertices.

In addition:
- Cleaned up code: removed unnecessary calls to `convex_hull`
- Corrected documentation for `convex_hull`.

Closes #146

### CODE REVIEW

**Comment by mcara** [comment]
I also added a unit test that uses the real data provided by @stsci-hack but a different WCS object in order to reduce complexity of the test.

File: tweakwcs/wcsimage.py [hunk 1]
```diff
@@ -1678,6 +1702,11 @@ def cross(o, a, b):
 
     # convert x, y vertex coordinates to RA & DEC:
     ra, dec = wcs(ptx, pty)
+
+    # TODO: for strange reasons, occasionally ra[0] != ra[-1] and/or
+    #       dec[0] != dec[-1] (even though we close the polygon in the
+    #       previous two lines). Then SphericalPolygon fails because
+    #       points are not closed. Threfore we force it to be closed:
     ra[-1] = ra[0]
```
Comment (stsci-hack) [inline tweakwcs/wcsimage.py:1710]: Do you still want to simply replace the last result with the first entry in the output list?  This assumes that the 'ra[-1]' is essentially floating-point equivalent to `ra[0]` already.  Is this assumption ALWAYS correct, as it is not obvious from the code alone (at least to me)?  If this assumption can not be guaranteed to ALWAYS be correct, then this line is introducing a potentially major change to the shape of the output polygon.

**Comment by stsci-hack** [comment]
It's always better to address the root cause of a problem than deal with the aftermath (as I was doing in #146).  Since this PR was successfully tested against the dataset that prompted #146, I support this set of changes as opposed to using #146.   I still have one concern which it would be good to confirm is really OK in the code.

**Review by stsci-hack — COMMENTED** [review]
Please see the comment about how the polygon gets closed.

**Pull request merged**